### ISSUE TITLE
matplotlib 3.6.1 breaks animate.TSdiagram

### ISSUE THREAD

**Issue by MaceKuailv** [malmans2/oceanspy#349]
Running the test with matplotlib 3.6.2 or 3.7.1 will result in a error in the The second test of test animate. 

This is what caused the first few failures in #348.

Here is a minimal example of the test that failed. I am not able to reproduce this because I am currently experiencing some difficulties in upgrading matplotlib.

```
import oceanspy as ospy
import xarray as xr
from matplotlib.animation import FuncAnimation

# od = ospy.open_oceandataset.from_catalog('get_started',catalog_url='./catalog_xarray.yaml')
od = ospy.open_oceandataset.from_catalog('get_started')
from oceanspy.animate import TS_diagram


anim = TS_diagram(
        od,
        cutout_kwargs={"ZRange": 0},
        colorName="Temp",
        Tlim=[0,1],
        Slim=[0,1],
        cmap_kwargs={"robust": True},
    )
# anim = TS_diagram(
#         od,
#         cutout_kwargs=None,
#         colorName=None,
#         Tlim=None,
#         Slim=None,
#         cmap_kwargs=None,
#     )
assert isinstance(anim, FuncAnimation)
```

**Comment by MaceKuailv** [comment]
The test pass when using 3.5.2. So somewhere between 3.5.2 and 3.6.2 lies the change that breaks this test. Although this result in a pin on matplotlib again, but it is a totally different issue from #269.

**Comment by MaceKuailv** [comment]
It turns out not all dataset have this issue. 
To reproduce this issue, one sort of have to use the test Dataset:
```
Datadir = "../oceanspy/oceanspy/tests/Data/"

# Test oceandataset
od = ospy.open_oceandataset.from_netcdf("{}MITgcm_rect_nc.nc" "".format(Datadir))
```
BTW, matplotlib 3.6.1 also break this test.
The output (There is no error raised) is:
```
Opening dataset from [../oceanspy/oceanspy/tests/Data/MITgcm_rect_nc.nc].
Cutting out the oceandataset.
Computing potential density anomaly using the following parameters: {'eq_state': 'jmd95'}.
  0%|                                                                                                                                                                               | 0/17 [00:00<?, ?it/s]Traceback (most recent call last):
  File "/home/wjiang33/oceanspy_dvlp/reproduce/animateIssue.py", line 13, in <module>
    anim = TS_diagram(
  File "/home/wjiang33/oceanspy_dvlp/oceanspy/oceanspy/animate.py", line 433, in TS_diagram
    anim = _create_animation(
  File "/home/wjiang33/oceanspy_dvlp/oceanspy/oceanspy/animate.py", line 112, in _create_animation
    _display(_HTML(anim.to_html5_video()))
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/animation.py", line 1266, in to_html5_video
    self.save(str(path), writer=writer)
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/animation.py", line 1068, in save
    anim._init_draw()  # Clear the initial frame
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/animation.py", line 1706, in _init_draw
    self._draw_frame(frame_data)
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/animation.py", line 1728, in _draw_frame
    self._drawn_artists = self._func(framedata, *self._args)
  File "/home/wjiang33/oceanspy_dvlp/oceanspy/oceanspy/animate.py", line 100, in animate
    plot_func(od, **func_kwargs)
  File "/home/wjiang33/oceanspy_dvlp/oceanspy/oceanspy/plot.py", line 295, in TS_diagram
    sc = ax.scatter(S.values.flatten(), T.values.flatten(), c=c, **kwargs)
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/__init__.py", line 1423, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/axes/_axes.py", line 4530, in scatter
    self._parse_scatter_color_args(
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/axes/_axes.py", line 4328, in _parse_scatter_color_args
    and isinstance(cbook._safe_first_finite(c), str)))
  File "/home/wjiang33/miniconda3/envs/ospy_tests/lib/python3.10/site-packages/matplotlib/cbook/__init__.py", line 1749, in _safe_first_finite
    return next(val for val in obj if safe_isfinite(val))
StopIteration
  0%|                                             
```

**Comment by MaceKuailv** [comment]
I have now confirmed that changes between 3.6.0 and 3.6.1 breaks the test. No luck in finding the root cause.

**Comment by Mikejmnez** [comment]
Thanks for looking into this @MaceKuailv . I am spending some time in oceanspy this week working on a (draft PR) that I have made good progress related to mooring and I will also take a look into this since I am a bit familiar at least with horizontal animations.

**Comment by Mikejmnez** [comment]
The error had to do with the way cutout was being defined before plotting/animating.

In the example above (first comment), the choices
```python
cutout_kwargs={"ZRange": 0},
        colorName="Temp",
        Tlim=[0,1],
        Slim=[0,1],
```
imply the limits for plotting on Temp and Salinity are 0 to 1, and only surface data is plotted. At this depth level all salinity and temperature values are greater that the (plotting) limits. As a result, the following lines (oceanspy.plot.py lines 283-286)
```python
color = color.where(np.logical_and(T > min(Tlim), T < max(Tlim)))
color = color.where(np.logical_and(S > min(Slim), T < max(Slim)))
color = color.stack(all_dims=color.dims)
c = color.values
```
yield the (plotting) variable `c` all nans (`c` is `Temp` here). Hence, the following line in oceanspy.plot.py
```python
c = ax.scatter(S.values.flatten(), T.values.flatten(), c=c, **kwargs)
```
cases error for `matplotlib >3.6.0`. For older versions of matplotlib, even though `c` was all nans, it still produced the plot (at least the contours with correct axes) and there was no error.

Note that the error is true for both `oceanspy.animate.TS_diagram` and `oceanspy.plot.TS_diagram`.

An easy way to fix this was to change the range of Tlim and Slim in `test_animate.py.` However,  I took a look at `test_plot.py`  (lines >70) and the same limits are being used for plotting `TS_diagram` with the same dataset and there there is no error there! I found out the difference was the cutout parameters. 

In `test_plot.py`, the cutout is defined by `XRange`, and `ZRange= None` (and so all depth levels are considered). This allows for `c` to have some non-nan values and therefore no error :).
